projects/packages/backup-helper-script-manager/src/class-throw-on-errors.php

```diff
@@ -0,0 +1,497 @@
+// phpcs:disable WordPress.PHP.IniSet.display_errors_Disallowed
+
+namespace Automattic\Jetpack\Backup;
+
+use Exception;
+
+/**
+ * Wrappers for functions which throw an exception on errors and warnings instead of silently continuing operation.
+ *
+ * PHP is pretty lax with error reporting when doing I/O operations, e.g. a typical I/O helper function returns false,
+ * null, and/or emits a warning, but the whole PHP application continues operation. It's for the caller of the I/O
+ * helper function to test the returned result of the function, and notice + act upon I/O errors.
+ *
+ * We really want to know about each and every I/O error. Therefore, this class provides wrappers for some common
+ * (mostly) I/O operations that throw an exception on errors instead of just returning false, null, or something else.
+ * This wrapper class treats warnings as errors too.
+ *
+ * Given that static method names are similar to the ones used by PHP, they're prefixed with "t_" to not erroneously
+ * trigger various security scanners.
+ */
+class Throw_On_Errors {
+
+	/**
+	 * Execute a callable, throw an exception (together with a descriptive label) on PHP warnings / errors.
+	 *
+	 * @param callable $callable Callable to execute.
+	 * @param string   $label Label to add to the thrown exception to clarify what was attempted.
+	 *
+	 * @return mixed Callable's return value, if any.
+	 * @throws Exception On warnings thrown by the callable.
+	 * @noinspection PhpUnusedParameterInspection
+	 */
+	private static function throw_on_warnings( $callable, $label ) {
+		$old_error_reporting = error_reporting( - 1 );
+		$old_display_errors  = ini_set( 'display_errors', 'stderr' );
+
+		// phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_set_error_handler
+		set_error_handler(
+		/**
+		 * Temporary error handler.
+		 *
+		 * @see https://php-legacy-docs.zend.com/manual/php5/en/function.set-error-handler
+		 * @see https://www.php.net/manual/en/function.set-error-handler.php
+		 *
+		 * @param int $errno Level of the error raised.
+		 * @param string $errstr Error message.
+		 * @param string|null $errfile Filename that the error was raised in.
+		 * @param int|null $errline Line number where the error was raised.
+		 * @param array|null $errcontext Deprecated, unused.
+		 *
+		 * @return mixed
+		 * @throws Exception
+		 */
+			// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+			function ( $errno, $errstr, $errfile = null, $errline = null, $errcontext = null ) {
+				throw new Exception( "$errstr (file: $errfile; line: $errline)" );
+			}
+		);
+
+		$result        = null;
+		$error_message = null;
+		if ( PHP_MAJOR_VERSION >= 7 ) {
+			// On PHP 7.x, all Exception and Error are subclasses of Throwable.
+			try {
+				$result = $callable();
+			} catch ( Throwable $throwable ) {
+				$error_message = $throwable->getMessage();
+			} catch ( Exception $exception ) {
+				$error_message = $exception->getMessage();
```

Comment:
This will never be reached, since the previous catch for Throwable will always do so instead.

This is the kind of code you'd write to also support PHP 5, but for some reason you're only running it when `PHP_MAJOR_VERSION >= 7`.

Reply:
Thanks, removed PHP 5.x hacks in 202ce2b.